components/button/BaseButton.tsx

```diff
@@ -23,17 +46,13 @@ export const BaseButton = (props: Props) => {
       onClick={props.onClick}
       type="button"
     >
-      {props.text && props.text !== "" && props.position === "left" && (
-        <div className="my-auto mr-2">
-          {props.text}
-        </div>
-      )}
-      {props.icon && React.cloneElement(props.icon, {
-        disabled: props.disabled,
-      })}
-      {props.text && props.text !== "" && (!props.position || props.position === "right") && (
-        <div className="my-auto ml-2">
-          {props.text}
+      {props.effect === true && props.text ? (
+        <EffectButton value={props.text}>
+          {render(props)}
+        </EffectButton>
```

Comment:
This is basically how we can use the `EffectButton` anywhere. The injected value is watched, and when it changes the component attaches the effects class name to the div wrapper, which animates the font color of the children through CSS inheritance.